src/Pages/Register/Register.tsx

```diff
@@ -1,10 +1,191 @@
-import React from 'react'
-import './Register.module.scss'
+import {
+  Box,
+  Button,
+  FormControl,
+  Grid,
+  Input,
+  InputLabel,
+  OutlinedInput,
+  Typography,
+} from "@mui/material";
+import axios from "axios";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import baseUrl from "../../utils/Custom/Custom";
+import "./Register.module.scss";
 
 const Register = () => {
-  return <>
-    <div>Register</div>
-  </>
-}
+  const navigate = useNavigate();
 
-export default Register
\ No newline at end of file
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const onSubmit = (data) => {
+    data.role = "user";
+    const addFormData = new FormData();
+    addFormData.append("userName", data["userName"]);
+    addFormData.append("email", data["email"]);
+    addFormData.append("password", data["password"]);
+    addFormData.append("confirmPassword", data["confirmPassword"]);
+    addFormData.append("phoneNumber", data["phoneNumber"]);
+    addFormData.append("country", data["country"]);
+    addFormData.append("role", data["role"]);
+    addFormData.append("profileImage", data["profileImage"][0]);
+
+    axios
+      .post(`${baseUrl}/api/v0/portal/users`, addFormData)
+      .then((res) => {
+        console.log("data", data);
+        toast.success("created successfully");
+        navigate("/login");
+      })
+      .catch((err) => {
+        console.log(err.response.data.message);
+        toast.error(err.message);
```

Comment:
This returns unuseful messages to the users such as error 401, so you need to change it to the message that shows the user good feedback like what is going wrong.